### DISCUSSION TITLE
UI Makeover

### DISCUSSION THREAD

**Discussion by sinamics** [blakeblackshear/frigate#1715] [Ideas]
I have never been 100% satisfied with the ui layout, so i have started a complete makeover for Frigate. 
This is a radical change and i guess would not be accepted as a PR, but who knows, maybe.. 

So before spending **100 hrs** on completing this project, i just want to know what you think, and if this is something that could be considered merged when completed. This will for sure break dev rules in terms of colors and button styles ect.

Features:
- Build UI much like Blue Iris.
- Bottom timeline.
- Events list in sidebar with hover : autoplay ++. 
- Compress as much information as possible on a single page.
- All functionalities will be remained.

Codebase:
- Convert everything to typescript.
- Not gonna use any 3rd party packages than already used.

Current layout. This is just a markup! Components will be much better styled and organized.
![makeover](https://i.ibb.co/441J0YJ/frigate-makeover.jpg)

**Comment by blakeblackshear** [comment]
There are some good ideas here. I'm not opposed to a major overhaul, but I don't want to simply mirror the UI from Blue iris. To me it has always felt very outdated, but I haven't used it much.

My preference would be to start with some use cases described in user stories.

**Reply by sinamics** [reply]
Thx.
I never used BI myself either, just like the way dashboard is organized (seen from images and YT). I will gather some use cases as suggested.
This is just a fun side project, and who knows, it might turn into something good in the end.

**Comment by trythatagain** [comment]
At the moment I'm still using Shinobi as I prefer the UI. I've not tried 0.9.0 yet, so when that gets released I might change my mind.
But I must say I do like the UI of Shinobi. 

I'm looking forward to see if you carry on with the project and perhaps adding some requests etc. Good luck with it!

**Comment by grimmaldus** [comment]
I second to Autoreplay feature. It worked more intuitive in 0.8.4 than in 0.9.1.
I truly hate to do extra clicks especially on mobile devices with one hand when I want to quickly checkout what's going on.

**Comment by msolomos** [comment]
For me something that must be considered is to give the option, for users, to change the layout. 
For example from one column, use multiple columns.  And give this option to the user. 

from this 
![image](https://github.com/blakeblackshear/frigate/assets/72358278/6dd66525-953a-494c-a4a4-c83fc21e0a0f)

to this

![image](https://github.com/blakeblackshear/frigate/assets/72358278/3faf797b-8f34-48a3-8c32-049aabab8b46)

Is not hard to make and it will help many users with many snapshots per day ....to find easier what they are looking for...

Please excuse my example but I couldn't find something similar to show ...
Here is the link if you want to take a look
https://demo.cs-cart.gr/electronics/computers/

**Comment by banthungprong** [comment]
The UI will be better in the next version. You can see these changes already in the dev-Version. This is already very nice.

![Screenshot from 2024-02-04 11-22-36](https://github.com/blakeblackshear/frigate/assets/59219161/754ef9e7-30ac-48b6-8e0a-10a76df32a4f)
